bindings/wasm/src/secret_manager.rs

```diff
@@ -6,58 +6,52 @@ use std::sync::Arc;
 use iota_sdk_bindings_core::{
     call_secret_manager_method as rust_call_secret_manager_method,
     iota_sdk::client::secret::{SecretManager, SecretManagerDto},
-    Response, SecretManagerMethod,
+    Response,
 };
 use tokio::sync::RwLock;
-use wasm_bindgen::{prelude::wasm_bindgen, JsCast, JsValue};
-use wasm_bindgen_futures::future_to_promise;
+use wasm_bindgen::{prelude::wasm_bindgen, JsError, JsValue};
 
-use crate::PromiseString;
+use crate::{build_js_error, map_err};
 
 /// The SecretManager method handler.
 #[wasm_bindgen(js_name = SecretManagerMethodHandler)]
 pub struct SecretManagerMethodHandler {
-    pub(crate) secret_manager: Arc<RwLock<SecretManager>>,
+    pub(crate) inner: Arc<RwLock<SecretManager>>,
```

Comment:
For this and the others, if there is only one field I'd make them unit structs

Reply:
done